ludwig/visualize.py

```diff
@@ -37,100 +38,729 @@
-        output_directory=None,
-        file_format='pdf',
-        **kwargs
+def validate_conf_treshholds_and_probabilities_2d_3d(
+        probabilities, treshhold_fields
 ):
-    if len(training_statistics) < 1:
-        logger.error('No training_statistics provided')
-        return
-
-    filename_template = None
-    if output_directory:
-        filename_template = os.path.join(
-            output_directory,
-            'learning_curves_{}_{}.' + file_format
+    """Ensure probabilities and treshhold fields arrays have two members each.
+
+    :param probabilities: List of probabilities per model
+    :param threshhold_fields: List of threshhold fields per model
+    :raise: RuntimeError
+    """
+    validation_mapping = {
+        'probabilities': probabilities,
+        'treshhold_fields': treshhold_fields
+    }
+    for item, value in validation_mapping.items():
+        item_len = len(value)
+        if not item_len == 2:
+            exception_message = 'Two {} should be provided - {} was given.'. \
+                format(item, item_len)
+            logging.error(exception_message)
+            raise RuntimeError(exception_message)
+
+
+def load_data_for_viz(load_type, model_file_statistics, **kwargs):
+    """Load model file data in to list of .
+
+    :param load_type: type of the data loader to be used.
+    :param model_file_statistics: JSON file or list of json files containing any
+           model experiment stats.
+    :return List of training statistics loaded as json objects.
+    """
+    supported_load_types = dict(load_json=load_json,
+                                load_from_file=partial(
+                                    load_from_file,
+                                    dtype=kwargs.get('dtype', int),
+                                    ground_truth_split=kwargs.get(
+                                        'ground_truth_split', 2)
+                                )
+                                )
+    loader = supported_load_types[load_type]
+    try:
+        stats_per_model = [loader(stats_f)
+                           for stats_f in
+                           model_file_statistics]
+    except (TypeError, AttributeError):
+        logging.exception(
+            'Unable to open model statistics file {}!'.format(
+                model_file_statistics
+            )
         )
+        raise
+    return stats_per_model
+
+def convert_to_list(item):
+    """If item is not list class instance or None put inside a list.
+
+    :param item: object to be checked and converted
+    :return: original item if it is a list instance or list containing the item.
+    """
+    return item if item is None or isinstance(item, list) else [item]
 
-    training_statistics_per_model_name = [load_json(learning_stats_f)
-                                          for learning_stats_f in
-                                          training_statistics]
 
+def validate_visualisation_prediction_field_from_train_stats(
```

Comment:
please replace every occurrence of "visualisation" with "visualization", adopting the american english convention ;)

Reply:
Done.